kythe/rust/extractor/src/bin/extractor.rs

```diff
@@ -0,0 +1,230 @@
+    }
+
+    // Grab the build target's output path
+    let build_output_path: &String = spawn_info
+        .get_output_file()
+        .get(0)
+        .ok_or_else(|| anyhow!("Failed to get output file from spawn info"))?;
+
+    // Add save analysis to kzip
+    let save_analysis_path: String = analysis_path_string(&build_output_path, &tmp_dir.path())?;
+    kzip_add_required_input(&save_analysis_path, &mut kzip, &mut required_input)?;
+
+    // Create the IndexedCompilation and add it to the kzip
+    let indexed_compilation = create_indexed_compilation(
+        rust_source_files,
+        compiler_arguments,
+        build_output_path.to_string(),
+        required_input,
+    );
+    let mut indexed_compilation_bytes: Vec<u8> = Vec::new();
+    indexed_compilation
+        .write_to_vec(&mut indexed_compilation_bytes)
+        .with_context(|| "Failed to serialize IndexedCompilation to bytes".to_string())?;
+    let indexed_compilation_digest = sha256digest(&indexed_compilation_bytes);
+    kzip_add_file(
+        format!("pbunits/{}", indexed_compilation_digest),
+        &indexed_compilation_bytes,
+        &mut kzip,
+    )?;
+
+    println!("Generated {:?}", &config.output_path);
+    Ok(())
+}
+
+// Inspired by https://gist.github.com/Igosuki/27e18369fec5a05d4019ac5d321e1779
+const SPAWN_INFO: protobuf::ext::ExtFieldOptional<
+    ExtraActionInfo,
+    protobuf::types::ProtobufTypeMessage<SpawnInfo>,
+> = protobuf::ext::ExtFieldOptional { field_number: 1003, phantom: ::std::marker::PhantomData };
+
+/// Attempt to parse ExtraActionInfo protobuf from file_path and extract the
+/// SpawnInfo extension
+fn get_spawn_info(file_path: impl AsRef<Path>) -> Result<SpawnInfo> {
+    let mut file = File::open(file_path).context("Failed to open extra action file")?;
+
+    let mut file_contents_bytes = Vec::new();
+    file.read_to_end(&mut file_contents_bytes).context("Failed to read extra action file")?;
+
+    let extra_action = protobuf::parse_from_bytes::<ExtraActionInfo>(&file_contents_bytes)
+        .context("Failed to parse extra action protobuf")?;
+
+    SPAWN_INFO.get(&extra_action).ok_or_else(|| anyhow!("SpawnInfo extension missing"))
+}
+
+/// Create an IndexedCompilation protobuf from the supplied arguments
+///
+/// * `source_files` - The names of the source files required by the build
+///   target
+/// * `arguments` - The arguments used to compile the target
+/// * `build_output_path` - The output path of the build target
+/// * `required_input` - The generated data for the CompilationUnit
+///   `required_input` field
+fn create_indexed_compilation(
+    source_files: Vec<String>,
+    arguments: Vec<String>,
+    build_output_path: String,
+    required_input: Vec<CompilationUnit_FileInput>,
+) -> IndexedCompilation {
+    let mut compilation_unit = CompilationUnit::new();
+
+    let mut vname = VName::new();
+    // TODO(Arm1stice): Determine proper Corpus/Root/Path for VName
+    vname.set_corpus("kythe".into());
```

Comment:
The `kythe` corpus is reserved and shouldn't be used during extraction.  It should generally be read from a `KYTHE_CORPUS` environment variable.  The C++ indexer defaults to using the corpus of the main source file if the environment variable is unset or empty.

Reply:
What would you recommend as a temporary corpus until the vname_config functionality is implemented?

Reply:
I would read from the `KYTHE_CORPUS` environment variable unconditionally and probably emit an error if it is unset.